pkg/services/grafana-apiserver/auth/authorizer/provider.go

```diff
@@ -1,34 +1,65 @@
 package authorizer
 
 import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apiserver/pkg/authorization/authorizer"
 	"k8s.io/apiserver/pkg/authorization/union"
 
-	"github.com/grafana/grafana/pkg/services/grafana-apiserver/auth/authorizer/impersonation"
-	"github.com/grafana/grafana/pkg/services/grafana-apiserver/auth/authorizer/org"
-	"github.com/grafana/grafana/pkg/services/grafana-apiserver/auth/authorizer/stack"
+	orgsvc "github.com/grafana/grafana/pkg/services/org"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
-func ProvideAuthorizer(
-	orgIDAuthorizer *org.OrgIDAuthorizer,
-	orgRoleAuthorizer *org.OrgRoleAuthorizer,
-	stackIDAuthorizer *stack.StackIDAuthorizer,
-	cfg *setting.Cfg,
-) authorizer.Authorizer {
+var _ authorizer.Authorizer = (*GrafanaAuthorizer)(nil)
+
+type GrafanaAuthorizer struct {
+	apis map[string]authorizer.AuthorizerFunc
```

Comment:
it might be more flexible to use `authorizer.Authorizer` in place of `authorizer.AuthorizerFunc` since it works for both types of authorizers (struct and function)